common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java

```diff
@@ -218,4 +224,184 @@ public void testGetErrorMessageWhenHiddenNull() {
+            );
+        parameters.put("context", "${parameters.text.toString()}");
+
+        List<String> prefixes = collectToStringPrefixes(parameters);
+        List<String> expectPrefixes = new ArrayList<>();
+        expectPrefixes.add("text");
+        expectPrefixes.add("context");
+        expectPrefixes.add("history");
+        assertEquals(prefixes, expectPrefixes);
+    }
+
+    /**
+     * Tests the parseParameters method with a map containing a list of strings as the value
+     * for the "context" key. Verifies that the method correctly processes the list and adds
+     * the processed value to the map with the expected key. Also tests the string substitution
+     * using the processed values.
+     */
+    @Test
+    public void testParseParametersListToString() {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("prompt", "answer question based on context: ${parameters.context.toString()}");
+        ArrayList<String> listOfDocuments = new ArrayList<>();
+        listOfDocuments.add("document1");
+        parameters.put("context", toJson(listOfDocuments));
+
+        parseParameters(parameters);
+        System.out.println(parameters);
+        assertEquals(parameters.get("context" + TO_STRING_FUNCTION_NAME), "[\\\"document1\\\"]");
+
+        String requestBody = "{\"prompt\": \"${parameters.prompt}\"}";
+        StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
+        requestBody = substitutor.replace(requestBody);
+        System.out.println(requestBody);
+        assertEquals(requestBody, "{\"prompt\": \"answer question based on context: [\\\"document1\\\"]\"}");
+    }
+
+    /**
+     * Tests the parseParameters method with a map containing a list of strings as the value
+     * for the "context" key, and the "prompt" value containing escaped characters. Verifies
+     * that the method correctly processes the list and adds the processed value to the map
+     * with the expected key. Also tests the string substitution using the processed values.
+     */
+    @Test
+    public void testParseParametersListToStringWithEscapedPrompt() {
+        Map<String, String> parameters = new HashMap<>();
+        parameters
+            .put(
+                "prompt",
+                "\\n\\nHuman: You are a professional data analyst. You will always answer question based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say I don't know. Context: ${parameters.context.toString()}. \\n\\n Human: please summarize the documents \\n\\n Assistant:"
+            );
+        ArrayList<String> listOfDocuments = new ArrayList<>();
+        listOfDocuments.add("document1");
+        parameters.put("context", toJson(listOfDocuments));
+
+        parseParameters(parameters);
+        System.out.println(parameters.get("context" + TO_STRING_FUNCTION_NAME));
```

Comment:
Remove the test code using `System.out.println`?

Reply:
removed. thanks!

Reply:
can you please review again? @ylwu-amzn